### PR TITLE
Raise exception in Route constructor when no HTTP methods provided

### PR DESCRIPTION
**Pull request by weierophinney**
Routes _require_ one or more HTTP methods in order to be valid.

### CODE REVIEW

File: src/Route.php [hunk 1]
```diff
@@ -160,6 +160,12 @@ public function getOptions() : array
      */
     private function validateHttpMethods(array $methods) : array
     {
+        if (empty($methods)) {
```
Comment (webimpress) [inline src/Route.php]: We can have here just `! $methods`

File: test/Middleware/PathBasedRoutingMiddlewareTest.php [hunk 2]
```diff
@@ -178,7 +178,7 @@ public function testCommonHttpMethodsAreExposedAsClassMethodsAndReturnRoutes($me
     public function testCreatingHttpRouteMethodWithExistingPathButDifferentMethodCreatesNewRouteInstance()
     {
         $this->router->addRoute(Argument::type(Route::class))->shouldBeCalledTimes(2);
-        $route = $this->middleware->route('/foo', $this->noopMiddleware, []);
+        $route = $this->middleware->route('/foo', $this->noopMiddleware, ['POST']);
```
Comment (webimpress) [inline test/Middleware/PathBasedRoutingMiddlewareTest.php]: Maybe we should use `RequestMethod::METHOD_POST` (for consistency)

File: test/RouteTest.php [hunk 3]
```diff
@@ -71,13 +71,13 @@ public function testRouteCanMatchMethod()
 
     public function testRouteHeadMethodIsNotAllowedByDefault()
     {
-        $route = new Route('/foo', $this->noopMiddleware, []);
+        $route = new Route('/foo', $this->noopMiddleware, ['GET']);
```
Comment (webimpress) [inline test/RouteTest.php]: also here and in following tests

File: src/Route.php (hunk 1 above)
Comment (weierophinney) [reply src/Route.php]: Considering the argument is typed as an array, this makes it more semantically clear that we're checking to see if we have an _empty_ array.

Comment (webimpress) [reply src/Route.php]: the result will be exactly the same and everywhere we are using just `!`, but fine - whatever you prefer, both works the same.

Comment (weierophinney) [reply src/Route.php]: I'm aware the results will be the same. Readability and intent are important, however!

**Pull request merged**